File: frontend/src/app/core/services/disponibilidad.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { ApiResponse } from '../models/api-response.model';

export interface Disponibilidad {
  id: number;
  proveedorId: number;
  servicioId: number;
  servicioNombre: string;
  diaSemana: number;
  diaSemanaNombre: string;
  horaInicio: string;
  horaFin: string;
  activo: boolean;
}

@Injectable({ providedIn: 'root' })
export class DisponibilidadService {
  constructor(private api: ApiService) {}

  getByProveedorServicio(proveedorId: number, servicioId: number): Observable<ApiResponse<Disponibilidad[]>> {
    return this.api.get(`/disponibilidades/by-proveedor-servicio/${proveedorId}/${servicioId}`);
  }

  getByServicio(servicioId: number): Observable<ApiResponse<Disponibilidad[]>> {
    return this.api.get(`/disponibilidades/by-servicio/${servicioId}`);
  }

  create(data: DisponibilidadRequest): Observable<ApiResponse<Disponibilidad>> {
    return this.api.post('/disponibilidades', data);
  }

  update(id: number, data: DisponibilidadRequest): Observable<ApiResponse<Disponibilidad>> {
    return this.api.put(`/disponibilidades/${id}`, data);
  }

  delete(id: number): Observable<ApiResponse<void>> {
    return this.api.delete(`/disponibilidades/${id}`);
  }
}

export interface DisponibilidadRequest {
  proveedorId: number;
  servicioId: number;
  diaSemana: number;
  horaInicio: string;
  horaFin: string;
}

File: services/provider-service/src/main/java/bo/capital/tec/pet/modules/proveedor/service/DisponibilidadService.java
package bo.capital.tec.pet.modules.proveedor.service;

import bo.capital.tec.pet.common.api.PagedResponse;
import bo.capital.tec.pet.modules.proveedor.dto.DisponibilidadRequestDTO;
import bo.capital.tec.pet.modules.proveedor.dto.DisponibilidadResponseDTO;

import java.util.List;

public interface DisponibilidadService {
    DisponibilidadResponseDTO create(DisponibilidadRequestDTO dto);
    DisponibilidadResponseDTO getById(Long id);
    PagedResponse<DisponibilidadResponseDTO> getAll(Long proveedorId, Long servicioId, int page, int size);
    List<DisponibilidadResponseDTO> getByProveedorAndServicio(Long proveedorId, Long servicioId);
    List<DisponibilidadResponseDTO> getByServicio(Long servicioId);
    DisponibilidadResponseDTO update(Long id, DisponibilidadRequestDTO dto);
    void delete(Long id);
}
